ens/main.py

```diff
@@ -111,14 +151,16 @@ def setup_address(self, name, address=default, transact={}):
         :raises InvalidName: if ``name`` has invalid syntax
         :raises UnauthorizedError: if ``'from'`` in `transact` does not own `name`
         """
+        if transact is None:
+            transact = cast(TxDict, {})
         owner = self.setup_owner(name, transact=transact)
         self._assert_control(owner, name)
         if is_none_or_zero_address(address):
             address = None
         elif address is default:
             address = owner
         elif is_binary_address(address):
-            address = to_checksum_address(address)
+            address = cast(Address, to_checksum_address(address))
```

Comment:
This seems wrong too.  I thought that `eth-utils` returned properly typed values from these.